transport/src/commonMain/kotlin/com/genesys/cloud/messenger/transport/core/MessagingClientImpl.kt

```diff
@@ -63,12 +67,12 @@ internal class MessagingClientImpl(
     override val conversation: List<Message>
         get() = messageStore.getConversation()
 
-    @Deprecated("Use the connectToSession() instead", ReplaceWith("connectToSession()"))
+    @Deprecated("Use the connect(shouldConfigure: Boolean) instead", ReplaceWith("connect(shouldConfigure: Boolean)"))
     @Throws(IllegalStateException::class)
     override fun connect() {
         log.i { "connect()" }
-        check(currentState is State.Closed || currentState is State.Idle || currentState is State.Error) { "MessagingClient state must be Closed, Idle or Error" }
-        currentState = State.Connecting
+        check(stateMachine.canConnect()) { "MessagingClient state must be Closed, Idle or Error" }
```

Comment:
Do you mean something like:

StateMachine:

    @Throws(IllegalArgumentException::class)
    override fun onConnect() {
        check(canConnect()) { "MessagingClient state must be Closed, Idle or Error" }
        currentState = if (isReconnecting()) State.Reconnecting else State.Connecting
    }
    
And then MessagingClientImpl will call:

    @Throws(IllegalStateException::class)
    override fun connect() {
        stateMachine.onConnect()
        webSocket.openSocket(socketListener)
    }
    
    
**OR**:
StateMachine:

@Throws(IllegalArgumentException::class)
internal fun StateMachine.canConnect() =
    check(currentState is State.Closed || currentState is State.Idle || currentState is State.Error || currentState is State.Reconnecting) { "MessagingClient state must be Closed, Idle or Error" }
    
MessagingClient:

    @Throws(IllegalStateException::class)
    override fun connect() {
        stateMachine.canConnect()
        stateMachine.onConnect()
        webSocket.openSocket(socketListener)
    }

Reply:
I was meaning the first of your two examples. My reasoning is that it one should probably not have to always ask StateMachine if it can connect before calling its onConnect transition. I think it would be cleaner to have the check inside onConnect.